dot/core/service.go

```diff
@@ -149,6 +152,42 @@ func (s *Service) StorageRoot() (common.Hash, error) {
 
 // HandleBlockImport handles a block that was imported via the network
 func (s *Service) HandleBlockImport(block *types.Block, state *rtstorage.TrieState, announce bool) error {
+	parentHash := block.Header.ParentHash
+	if parentHash != s.blockState.GenesisHash() {
+		parentHeader, err := s.blockState.GetHeader(parentHash)
+		if err != nil {
+			return fmt.Errorf("getting parent header: %w", err)
+		}
+
+		parentEpoch, err := s.epochState.GetEpochForBlock(parentHeader)
+		if err != nil {
+			return fmt.Errorf("getting epoch for parent block: %w", err)
+		}
+
+		currentBlockEpoch, err := s.epochState.GetEpochForBlock(&block.Header)
+		if err != nil {
+			return fmt.Errorf("getting epoch for current block: %w", err)
+		}
+
+		// if epoch was skipped then we should change the current
+		// epoch descriptor mapping to use the actual epoch,since
+		// was expected to have a block on `parentEpoch + 1` but
+		// the descendant is more than one epoch foward
+		if currentBlockEpoch > (parentEpoch + 1) {
+			_, err := s.epochState.GetSkippedEpochDataRaw(parentEpoch+1,
+				currentBlockEpoch, &block.Header)
+			if err != nil {
+				return fmt.Errorf("updating skipped epoch data raw: %w", err)
+			}
+
+			_, err = s.epochState.GetSkippedConfigData(parentEpoch+1,
```

Comment:
I would say same here, its confusing to me to have a getter where we ignore the returned value